src/Api.php

```diff
@@ -4,46 +4,93 @@
 
 class Api
 {
-    public static $baseUrl = "https://api.razorpay.com/v1/";
+    protected static $baseUrl = 'https://api.razorpay.com/v1/';
 
-    public static $key = null;
+    protected static $key = null;
 
-    public static $secret = null;
+    protected static $secret = null;
+
+    /*
+     * App info is to store the Plugin/integration
+     * information
+     */
+    public static $appsDetails = null;
 
     const VERSION = '1.2.9';
 
     /**
      * @param string $key
      * @param string $secret
      */
-    function __construct($key, $secret)
+    public function __construct($key, $secret)
     {
         self::$key = $key;
         self::$secret = $secret;
-
-        // Add the version to all HTTP Requests
-        Request::addHeader('User-Agent', "Razorpay-PHP/".self::VERSION);
     }
 
     /*
      *  Set Headers
      *
      */
-    function setHeader($header, $value)
+    public function setHeader($header, $value)
     {
         Request::addHeader($header, $value);
     }
 
+    public function setAppDetails($title, $version = null)
```

Comment:
if setAppDetails is not called , then getAppDetailsUa will fail in Request.php as appsDetails is by default as null.
change default value to array()